### ISSUE TITLE
Is there a reason we are hard coding the storage driver type in the database.

### ISSUE THREAD

**Issue by rhatdan** [containers/libpod#1826]
I would want to be able to use multiple different storages in podman at the same time.  But the database refuses to allow me?

**Comment by rhatdan** [comment]
@mheon WHY?

**Comment by rhatdan** [comment]
I am running podman in my homedir, and I want to play with fuse-overlay, but I have to blow away all containers, since i ran originally with the vfs driver.

**Comment by mheon** [comment]
If you swap storage driver type with active containers, suddenly the
contents of your storage dirs become useless, and we lose whatever active
containers you have - you need to delete them and recreate them on the new
storage driver. Basically, we're trying to ensure database consistency by
making sure everyone using it can see and use the same set of containers in
c/storage

**Comment by mheon** [comment]
I think our basic answer for swapping to fuse-overlayfs is tell people to
blow away their database after saving all relevant containers...

**Comment by rhatdan** [comment]
Except that figuring out where your content is in your homedirs is difficult.  And podman info stops working.  Perhaps we could make the error indicate to them that changing your storage driver requires a reset of storage.

But I still would like to run with two different storages at the same time.

What if I want to experiment with KATA containers or use kata for certain workloads while I use runc for others.  KATA works much better with devicemapper, while I definitely want to use overlay with traditional containers.   Is there any reason we can not support this use case?

**Comment by vrothberg** [comment]
> [...] Perhaps we could make the error indicate to them that changing your storage driver requires a reset of storage.

Would changing the location of the storage work as well? If so, couldn't storage take care of that on its own and create per-driver databases?

**Comment by rhatdan** [comment]
Well if we moved the libpod directory under the storage-driver directory it could work.
@mheon Would this cause large issues?

**Comment by rhatdan** [comment]
Well we could allways create overlay-libpod.

And yes I expect the containers to disappear when the drive is changed.

**Comment by TomasTomecek** [comment]
I completely agree with Dan here. I'm trying to move from one driver to another right now and can't:
```
Could not get runtime: database graph driver name overlay does not match our graph driver name vfs: database configuration mismatch
```
If you care about running containers, you could easily print a warning that there are containers running with driver $x while driver $y is being used right now.

**Comment by mheon** [comment]
I think our problem with the proposed fix is that it's a breaking change
for everyone using Podman. Everyone, rootless and not, will lose all their
containers of we move the DB. We need a way of maintaining backwards
compatibility if we want to fix this.

**Comment by rhatdan** [comment]
I agree, I was thinking that the tool shold look to see if the old format exists and  matches the driver, then we copy/mv it to the new directory.  I will attempt to do this on the next pass.

**Comment by baude** [comment]
where are we on this?

**Comment by mheon** [comment]
@rhatdan was working on a PR to move the DB inside the storage driver-specific directories to allow use of multiple drivers. Unfortunately that's a breaking change, so I think it's stalled

**Comment by rhatdan** [comment]
https://github.com/containers/libpod/pull/1847
For some reason I can not get it to pass all tests...

**Comment by rhatdan** [comment]
I would still like to get this PR in.  But I have not figured out why rootless does not like it. Hoping that @giuseppe takes it over.

**Comment by rhatdan** [comment]
This PR Finally passes tests, but now need to work on the upgrade path.  If anyone wants to grab it, you are welcome to take it.

**Comment by baude** [comment]
do we still need this open?

**Comment by rhatdan** [comment]
I have had a PR for this open for many many months.  Post 1.5, I would like to finally get it merged.

**Comment by baude** [comment]
@rhatdan did you get your pr merged on this so we can close it ?

**Comment by mheon** [comment]
Nope.

**Comment by rhatdan** [comment]
PR Is still out there.

**Comment by rhatdan** [comment]
PR is still out there.
